### PR TITLE
Add description of a ddoc structure

### PR DESCRIPTION
**Pull request by tommikarkkainen**
## Overview

Describes the structure of a design document in a newbie-friendly way.

## Testing recommendations

Please check that the example I added is correct. I've tested it to the best of my abilities but would like someone else to take a look at it too.

## GitHub issue number

Related to #359 and #429 

## Checklist

- [ ] Update [rebar.config.script](https://github.com/apache/couchdb/blob/main/rebar.config.script) with the commit hash once this PR is rebased and merged

### CODE REVIEW

File: src/ddocs/ddocs.rst [hunk 1]
```diff
@@ -32,19 +32,47 @@ them extensively within each function:
 - :ref:`Database Security object <security_object>`
 - :ref:`Guide to JavaScript Query Server <query-server/js>`
 
-Creation
-========
+Creation and structure
```
Comment (nickva) [inline src/ddocs/ddocs.rst:35]: Minor nit: maybe use a capital `S` in `Structure` just for consistency with other titles. I don't actually remember the correct rule there but consistency would be better I think.

**Review by nickva — APPROVED** [review]
@tommikarkkainen thank you for your contribution.

I remember being tripped up by not being able to find complete examples of design documents when learning to use CouchDB

See the minor nit about title consistency though and then +1 and I'll can help merge it.

File: src/ddocs/ddocs.rst (hunk 1 above)
Comment (tommikarkkainen) [reply src/ddocs/ddocs.rst]: Well spotted! I just pushed the fix. Thank you.

**Comment by nickva** [comment]
@tommikarkkainen I merged the PR. Thank you again for your contribution!

**Pull request merged**